http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeServiceDbQueries.scala

```diff
@@ -306,411 +247,267 @@ object RuntimeServiceDbQueries {
+          Option[IP],
+          String,
+          Boolean,
+          RuntimeConfigRecord,
+          RuntimeName,
+          RuntimeStatus,
+          Option[WorkspaceId],
+          Option[(String, String)]
+        )
+        records
+          .groupBy(_._1)
+          .map { case _ -> (values: Seq[ListRuntimesRecord]) =>
+            val allLabels: LabelMap = Map(values.mapFilter {
+              case (_, _, _, _, _, _, _, _, _, _, _, _, _, _, labelPair: Option[(String, String)]) => labelPair
+            }: _*)
+            val patchInProgress: Boolean = values.exists((value: ListRuntimesRecord) =>
+              value match {
+                case (_, _, _, _, _, _, _, _, _, inProgress: Boolean, _, _, _, _, _) => inProgress
+              }
+            )
+            values.head match {
+              case (
+                    id,
+                    cloudContextDb,
+                    cloudProvider,
+                    creatorEmail,
+                    createdDate,
+                    destroyedDate,
+                    dateAccessed,
+                    hostIp,
+                    internalId,
+                    _,
+                    runtimeConfigRecord,
+                    runtimeName,
+                    status,
+                    workspaceId,
+                    _
+                  ) =>
+                val auditInfo = AuditInfo(
+                  creatorEmail,
+                  createdDate,
+                  if (destroyedDate == dummyDate) None else Some(destroyedDate),
+                  dateAccessed
+                )
+                val cloudContext = (cloudProvider, cloudContextDb) match {
+                  case (CloudProvider.Gcp, CloudContextDb(value)) =>
+                    CloudContext.Gcp(GoogleProject(value))
+                  case (CloudProvider.Azure, CloudContextDb(value)) =>
+                    val context = AzureCloudContext.fromString(value).fold(s => throw new SQLDataException(s), identity)
+                    CloudContext.Azure(context)
+                }
+                val proxyUrl = Runtime.getProxyUrl(
+                  Config.proxyConfig.proxyUrlBase,
+                  cloudContext,
+                  runtimeName,
+                  Set.empty,
+                  hostIp,
+                  allLabels
+                )
+                val runtimeConfig = runtimeConfigRecord.runtimeConfig
+                val samResourceId = RuntimeSamResourceId(internalId)
+
+                ListRuntimeResponse2(
+                  id,
+                  auditInfo = auditInfo,
+                  cloudContext = cloudContext,
+                  clusterName = runtimeName,
+                  labels = allLabels,
+                  patchInProgress = patchInProgress,
+                  proxyUrl = proxyUrl,
+                  runtimeConfig = runtimeConfig,
+                  samResource = samResourceId,
+                  status = status,
+                  workspaceId = workspaceId
+                )
```

Comment:
The formatted result! Doing all this explicitly instead of relying on some of the convenience structures in `ClusterComponent` because of my lack of familiarity with those; I didn't want to inadvertently introduce complexity.